sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/annotation/ForbidAnnotationElementValueCheck.java

```diff
@@ -0,0 +1,386 @@
+ * &lt;property name="forbiddenElementValueRegexp" value="unchecked"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Code
+ * </p>
+ *
+ * <pre>
+ * &#64;SuppressWarnings("unchecked")
+ * </pre>
+ * <p>
+ * To forbid any array-valued element, frobiddenElementValueRegexp option should be: "\{.*\}".
+ * </p>
+ * <p>
+ * Config
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="ForbidAnnotationElementValue"&gt;
+ * &lt;property name="annotationName" value="Endorsers"/&gt;
+ * &lt;property name="elementName" value="value"/&gt;
+ * &lt;property name="forbiddenElementValueRegexp" value="}\{.*\}"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * <p>
+ * Code
+ * </p>
+ *
+ * <pre>
+ * &#64;Endorsers({"Children", "Unscrupulous dentists"})<br>
+ * public class Lollipop { ... }
+ * </pre>
+ *
+ * @author <a href="mailto:drozzds@gmail.com"> Sergey Drozd </a>
+ * @author Richard Veach
+ */
+public class ForbidAnnotationElementValueCheck extends Check {
+    /** Message key. */
+    public static final String MSG_KEY = "annotation.forbid.element.value";
+
+    /** CharMatcher using to trimming quotes from Strings. */
+    private static final CharMatcher QUOTE_MATCHER = CharMatcher.is('\"');
+
+    /** Default forbidden annotation element name. */
+    private static final String DEFAULT_ELEMENT_NAME = "value";
+
+    /** Precompiled forbidden element value pattern. */
+    private Pattern forbiddenElementValueRegexp = Pattern.compile("^$");
+
+    /** Forbidden annotation name. */
+    private String annotationName;
+
+    /** Forbidden annotation element name. */
+    private String elementName;
+
+    /**
+     * Sets Annotation Name Check property.
+     *
+     * @param annotationName
+     *        THe annotation name.
+     */
+    public void setAnnotationName(String annotationName) {
+        this.annotationName = annotationName;
+    }
+
+    /**
+     * Sets Annotation Element Check property.
+     *
+     * @param elementName
+     *        The annotation element name.
+     */
+    public void setElementName(String elementName) {
+        this.elementName = elementName;
+    }
+
+    /**
+     * Sets Forbidden Element Value Pattern Check property.
+     *
+     * @param forbiddenElementValueRegexp
+     *        The forbidden element value pattern to set.
+     */
+    public void setForbiddenElementValueRegexp(String forbiddenElementValueRegexp) {
+        this.forbiddenElementValueRegexp = Pattern.compile(forbiddenElementValueRegexp);
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.ANNOTATION,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (getAnnotationName(ast).equals(annotationName)) {
+            if (DEFAULT_ELEMENT_NAME.equals(elementName) && isSingleElementAnnotation(ast)) {
+                log(getSingleElementWithForbiddenValue(ast), MSG_KEY, DEFAULT_ELEMENT_NAME,
+                        annotationName);

```

Comment:
please move getSingleElementWithForbiddenValue out of log.
Whole logic is in this method, it should not be hidden.
`log` line should be similar to next `log` line, by structure.